modules/data-catalog-tag-template/main.tf

```diff
@@ -0,0 +1,67 @@
+/**
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+locals {
+  _factory_tag_template = {
+    for f in fileset(var.factory_config_path, "**/*.yaml") :
+    trimsuffix(f, ".yaml") => yamldecode(file("${var.factory_config_path}/${f}"))
+  }
+
+  _tag_template_default = {
```

Comment:
is there a better way to handle defaults?

Reply:
I prefer:
```hcl
  is_required = try(fields.value["is_required"], false)
```
To:
```hcl
  is_required = try(fields.value["is_required"], local._tag_template_default.fields.is_required)
```

As you can see the default value exactly where it is used.

Reply:
I see, thanks. Updated.